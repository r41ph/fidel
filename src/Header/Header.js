import React from "react";
import "./Header.scss";

const Header = () => {
  return (
    <header className="f-header">
      <span className="f-header__title">Fidel API - Transactions List</span>
    </header>
  );
};

export default Header;
