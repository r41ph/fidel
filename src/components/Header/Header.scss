@import "../../assets/styles/colors.scss";

.f-header {
  &__title {
    color: $color-primary;
  }
}
