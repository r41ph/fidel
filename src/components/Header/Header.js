import React from "react";
import "./Header.scss";
import Logo from "../Logo/Logo";

const Header = () => {
  return (
    <header className="f-header">
      <Logo />
      <span className="f-header__title">Transactions</span>
    </header>
  );
};

export default Header;
