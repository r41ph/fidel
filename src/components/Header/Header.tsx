import React, { FunctionComponent } from "react";
import "./Header.scss";
import Logo from "../Logo/Logo";

const Header:FunctionComponent<{}> = () => {
  return (
    <header className="f-header">
      <Logo />
      <span className="f-header__title">Transactions</span>
    </header>
  );
};

export default Header;
