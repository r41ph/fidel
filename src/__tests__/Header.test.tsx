import React from "react";
import Header from "../components/Header/Header";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Header component", () => {
  it("contains header with text 'Transactions'", () => {
    const wrapper = shallow(<Header />);
    const text = wrapper.find(".f-header__title").text();
    expect(text).toEqual("Transactions");
  });

  it("matches the snapshop", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
