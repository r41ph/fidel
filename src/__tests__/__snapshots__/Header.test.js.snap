// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshop 1`] = `
<header
  className="f-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <span
    className="f-header__title"
  >
    Transactions
  </span>
</header>
`;
